internal/commands/branch/branch.go

```diff
@@ -0,0 +1,105 @@
+package branch
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	colorBranch = "\033[32m"
+	colorReset  = "\033[0m"
+)
+
+// CommandBranch lists git branches with numbered shortcuts.
+// The first line of output, when --branchlist is provided, will contain
+// a TAB separated list of branch names suitable for environment expansion.
+func CommandBranch() *cobra.Command {
+	var optsBranchlist bool
+
+	var branchCmd = &cobra.Command{
+		Use:   "branch",
+		Short: "Display numbered git branches",
+		Run: func(cmd *cobra.Command, args []string) {
+			branches := gitBranchOutput()
+			numbered, list := process(branches)
+
+			if optsBranchlist {
+				fmt.Println(strings.Join(list, "\t"))
+			}
+			fmt.Print(numbered)
+		},
+	}
+
+	branchCmd.Flags().BoolVarP(
+		&optsBranchlist,
+		"branchlist", "f", false,
+		"include machine-parseable branch list",
+	)
+
+	return branchCmd
+}
+
+func gitBranchOutput() []byte {
+	out, err := exec.Command("git", "branch", "--color=never").Output()
```

Comment:
It looks like there unfortunately isn't a `--porcelain` (guaranteed not to change across git versions) version of `git branch` output like there is with `git status`?  I'm wondering how to make the parsing resilient to multiple versions of git in the wild (including future versions).  Of course, once I get around to #33 that could be a good long term solution.  Another possibility while relying on CLI git, do you think it might make sense to use something like `--format` to specify a specific format? I'm not sure if that might be more resilient to change.

Reply:
Another edge case to consider might be whether there is a way to ensure the pager is disabled on a per command basis here, so git doesn't try to run its output through a pager if there more lines of branches than it thinks will fit on the user screen.  From the git branch documentation it appears that is the default, we might want to verify if it is automatically disabled somehow based on detecting whether STDOUT is a pipe or tty.